notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+
+
+
+## Geometrical Intuition
+
+Multiplying $X$ and $\hat{\theta}$ represents adding scalar multiples of the columns of $X$: $\vec{x}$ and $\vec{1}$, resulting in a linear combination. Here is an intuition:
+
+Let's take two generic vectors, $\vec{x}$ and $\vec{1}$, colored black. We use red to represent the new vector we are adding, and blue for the older linear combinations.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0], [0, 0], [1, 2], [1, -4], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_12_0.png)
+
+
+When $\hat{\theta_0} = 2$ and $\hat{\theta_1} = 0.5$, we add half of $x$ with two times the unit vector, resulting in a new vector.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0], [0, 0, 0], [1, 2, 3], [1, -4, 0], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_14_0.png)
+
+
+When $\hat{\theta_0} = 3$ and $\hat{\theta_1} = -1$, we add $3$ times the unit vector to $-1$ times $x$, resulting in yet another vector.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0, 0], [0, 0, 0, 0], [1, 2, 3, 1], [1, -4, 0, 7], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'blue', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_16_0.png)
+
+
+With $\hat{\theta_0} = -2$ and $\hat{\theta_1} = -1$, we get:
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [1, 2, 3, 1, -4], [1, -4, 0, 7, 2], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'blue', 'blue', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_18_0.png)
+
+
+With $\hat{\theta_0} = -4$ and $\hat{\theta_1} = 1$, we get:
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0], [1, 2, 3, 1, -4, -2], [1, -4, 0, 7, 2, -8], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'blue', 'blue', 'blue', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_20_0.png)
+
+
+This process can go on and on, showing that when we can use any scalar multiples, we can make an infinite number of linear combinations. However, infinite does not mean any--the linear combinations span a space defined by the original vectors. In this case, we would create a plane as you can start to see above. Formally, this illustrates how vectors $\vec{x}$ and $\vec{1}$ span a vector space.
```

Comment:
I still don't really understand what this section means :( We do not create a plane by adding more and more vectors above. x and 1 already span R2. No plane is being created, right?

Also what is the point of this geometric intuition

Reply:
adding more and more vectors is the intuition for x and 1 spanning r2. we can take this out if we assume people are solid in linalg

develop an intuition for why we care that $\hat{y}$ is a linear combination of the columns of $X$